cpp/dolfinx/fem/Form.h

```diff
@@ -49,30 +49,39 @@ struct integral_data
   /// @param id Domain ID.
   /// @param kernel Integration kernel.
   /// @param entities Entities to integrate over.
-  template <typename K, typename V>
+  /// @param enabled_coefficients Coefficients in form that are enabled in this
+  /// integral
+  template <typename K, typename V, typename Q>
     requires std::is_convertible_v<
                  std::remove_cvref_t<K>,
                  std::function<void(T*, const T*, const T*, const U*,
                                     const int*, const uint8_t*)>>
                  and std::is_convertible_v<std::remove_cvref_t<V>,
                                            std::vector<std::int32_t>>
-  integral_data(int id, K&& kernel, V&& entities)
+                 and std::is_convertible_v<std::remove_cvref_t<Q>,
+                                           std::vector<std::int8_t>>
+  integral_data(int id, K&& kernel, V&& entities, Q&& enabled_coefficients)
       : id(id), kernel(std::forward<K>(kernel)),
-        entities(std::forward<V>(entities))
+        entities(std::forward<V>(entities)),
+        enabled_coefficients(std::move(enabled_coefficients))
```

Comment:
`std::move` here is wrong - check how the other cases are handled.

Reply:
It is quite small, number of unique coefficients.